qualtran/bloqs/block_encoding/negate.py

```diff
@@ -0,0 +1,151 @@
+#  Copyright 2024 Google LLC
+#
+#  Licensed under the Apache License, Version 2.0 (the "License");
+#  you may not use this file except in compliance with the License.
+#  You may obtain a copy of the License at
+#
+#      https://www.apache.org/licenses/LICENSE-2.0
+#
+#  Unless required by applicable law or agreed to in writing, software
+#  distributed under the License is distributed on an "AS IS" BASIS,
+#  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#  See the License for the specific language governing permissions and
+#  limitations under the License.
+
+from functools import cached_property
+from typing import Dict, List, Set, Tuple, Union
+
+from attrs import frozen
+
+from qualtran import (
+    bloq_example,
+    BloqBuilder,
+    BloqDocSpec,
+    QAny,
+    Register,
+    Signature,
+    Soquet,
+    SoquetT,
+)
+from qualtran.bloqs.basic_gates import XGate, ZGate
+from qualtran.bloqs.block_encoding import BlockEncoding
+from qualtran.bloqs.block_encoding.lcu_select_and_prepare import PrepareOracle
+from qualtran.bloqs.bookkeeping.auto_partition import AutoPartition, Unused
+from qualtran.resource_counting import BloqCountT, SympySymbolAllocator
+from qualtran.symbolics import SymbolicFloat, SymbolicInt
+
+
+@frozen
+class Negate(BlockEncoding):
```

Comment:
Can we write something that absorbs an arbitrary phase instead of just `-1` ?

Reply:
nvm, I guess absorbing the negatives is sufficient since we are guaranteed that the LCU coefficients are real numbers and LCU can already work with arbitrary positive numbers ?

Reply:
I was also wondering this, do LCU hamiltonians always have real coeffs?

Reply:
I was pondering this. The only cases I've seen for block encodings have real coefficients. But I can change this to an arbitrary phase if we want to be more general.

Reply:
Thanks @anurudhp for pointing out that `GlobalPhase` works here to enable us to apply an arbitrary phase and also save the resource qubit.